src/handler.rs

```diff
@@ -105,17 +110,18 @@ impl Handler {
         }
     }
 
-    pub fn handle(
+    pub async fn handle(
         &mut self,
         request: PolymorphicRequest,
+        ctx: RequestContext,
     ) -> Result<Option<String>, String> {
         let method = request.method();
         let handler = match self.mapping.get(&method) {
             Some(h) => h,
             None => &self.default_handler,
         };
 
-        let resp = handler.handle(request)?;
+        let resp = handler.handle(request, ctx).await?;
```

Comment:
curiosity: why the `?` here?

Reply:
its for result values, basically it says if its Ok then resp gets that value otherwise return the err from the function